### DISCUSSION TITLE
Double entries when using ->query()->fetchAll()

### DISCUSSION THREAD

**Discussion by AndiLeni** [catfan/Medoo#1093] [Q&A] [answered]
Hello,

I have some putput from a query which I do not quite understand.
This code 
```php
$entries = $database->query('select see.id, see.name, see.location, see.lat, see.lon, data.temp, data.info, data.updated, data.lastaccess, data.website from see join data on see.id = data.see order by see.name asc')->fetchAll();
```
Gives me some weird results which contains both the column name as a key as well as an integer as a key.
````
40 => array:20 [
    "id" => "14"
    0 => "14"
    "name" => "Wörthsee"
    1 => "Wörthsee"
    "location" => "Starnberg"
    2 => "Starnberg"
    "lat" => "48.059925073695055"
    3 => "48.059925073695055"
    "lon" => "11.180303476687245"
    4 => "11.180303476687245"
    "temp" => "24,7"
    5 => "24,7"
    "info" => "xxxx"
    6 => "xxxx"
    "updated" => "25.06.2023 13:15"
    7 => "25.06.2023 13:15"
    "lastaccess" => "2023-06-25 16:53:14"
    8 => "2023-06-25 16:53:14"
    "website" => "xxxx"
    9 => "xxxx"
  ]
````
I am not certain whether I did something wrong or this is an issue. 
Do I need to disable this behavior somewhere?

Thanks and kind regards
Andi

**Comment by catfan** [comment, marked as answer]
The `query()` returns `PDOStatement` object, and you are calling its `fetchAll()`.

The result is correct with both index and key by default.

To get a different type of result, check out the parameters from the documentation.
https://www.php.net/manual/en/pdostatement.fetchall.php

**Reply by AndiLeni** [reply]
I wasn't aware that it returns a PDOStatement. Using FETCH_ASSOC solved the issue. Thank you for your answer!
